parser/src/powdr.lalrpop

```diff
@@ -694,6 +696,24 @@ PatternList: Vec<Pattern> = {
 
 // ---------------------------- Trait/Impl Declaration -----------------------------
 
+GenericTraitName: (SymbolPath, TraitScheme) = {
```

Comment:
I would expect this section to start with TraitDeclaration and GenericTraitName to go after TraitImplementation